modules/ps_mainmenu/ps_mainmenu.tpl

```diff
@@ -1,104 +1,140 @@
+    >
+      <ul class="menu__list">
+        {if $depth >= 1}
+          <li class="main-menu__title h5">{$parent.label}</li>
         {/if}
-      </li>
-      {if $node.children|count}
-        {$node.parent = $node}
-        {$node.expandId = $_expand_id}
-        {$children[] = $node}
-      {/if}
+        {foreach from=$nodes item=node}
+          <li 
+            class="{$node.type}{if $node.current} current{/if}{if $node.children|count} menu--childrens{/if}"
+            id="{$node.page_identifier}"
+          >
+            <a 
+              class="{if $depth>= 0}menu__link{/if}"
+              href="{$node.url}"
+              data-depth="{$depth}"
+              {if $node.open_in_new_window}target="_blank"{/if}
+            >
+            {$node.label}
+            </a>
+            {if $node.children|count}
+              {* Cannot use page identifier as we can have the same page several times *}
+              {assign var=_expand_id value=10|mt_rand:100000}
+              <span class="main-menu__toggle-child js-menu-open-child" data-target="{$_expand_id}">
+                <span data-target="#top_sub_menu_{$_expand_id}">
+                  <i class="material-icons rtl-flip">chevron_right</i>
+                </span>
+              </span>
+            {/if}
+          </li>
+          {if $node.children|count}
+            {$node.parent = $node}
+            {$node.expandId = $_expand_id}
+            {$children[] = $node}
+          {/if}
+        {/foreach}
+      </ul>
+    </nav>
+    {foreach from=$children item=child}
+      {mobileMenu 
+        nodes=$child.children 
+        depth=$child.children[0].depth 
+        parent=$child backTitle=$child.parent.label 
+        expandId=$child.expandId
+      }
     {/foreach}
-  </ul>
-</nav>
-
-{foreach from=$children item=child}
-  {mobileMenu nodes=$child.children depth=$child.children[0].depth parent=$child backTitle=$child.parent.label expandId=$child.expandId}
-{/foreach}
-{/if}
+  {/if}
 {/function}
 
 <div id="_desktop_menu" class="main-menu main-menu--desktop order-0 order-xl-1">
   <div class="d-none d-xl-block position-static js-menu-desktop">
-    {menu nodes=$menu.children}
+    {desktopMenu nodes=$menu.children}
   </div>
-
-  <button class="main-menu__toggler btn btn-unstyle d-xl-none me-2" type="button" data-bs-toggle="offcanvas"
-    data-bs-target="#mobileMenu" aria-controls="mobileMenu">
+  <button
+    class="main-menu__toggler btn btn-unstyle d-xl-none me-2" 
+    type="button" 
+    data-bs-toggle="offcanvas"
+    data-bs-target="#mobileMenu"
+    aria-controls="mobileMenu"
+  >
     <i class="material-icons">menu</i>
   </button>
 </div>
 
-<div class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas" tabindex="-1" id="mobileMenu"
-  aria-labelledby="mobileMenuLabel">
+<div
+  class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas" 
+  tabindex="-1" 
+  id="mobileMenu"
+  aria-labelledby="mobileMenuLabel"
```

Comment:
I guess no content is linked to that label, we should either use aria-label or use a good element to describe this canvas, wdyt?

Reply:
Yep!
It's just a container, I think something like this is enough:
`<div id="mobileMenu" class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas">`

Reply:
Probably, as the id will be taken in account when screen readers talks?

Reply:
`<div class="main-menu__offcanvas offcanvas offcanvas-start js-menu-canvas" id="mobileMenu">`

Reply:
I don't have any idea about this one!

Reply:
I guess yes